src/clj/puppetlabs/services/config/puppet_server_config_service.clj

```diff
@@ -26,7 +26,9 @@
           "Initializing with the following settings from core Puppet:\n%s"
           (ks/pprint-to-string puppet-config))
         (core/init-webserver! override-webserver-settings!
-                              (get-in tk-config [:webserver])
+                              (get-in tk-config
+                                      [:webserver :puppet-server]
+                                      (get-in tk-config [:webserver]))
                               puppet-config)
```

Comment:
Ah yes, I see:

https://github.com/puppetlabs/puppet-server/blob/master/src/clj/puppetlabs/services/legacy_routes/legacy_routes_service.clj#L26-26

That service should really have a dependency on the master service - otherwise that call is not really safe.

And, if we were to apply that principle to the current situation as well, we'd end up with a circular dependency - the master service already has a dependency on `:PuppetServerConfigService`.